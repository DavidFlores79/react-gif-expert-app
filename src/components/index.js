export * from './AddCategory';
export * from './InputComponent';
export * from './GifGrid';
export * from './GIfItem';