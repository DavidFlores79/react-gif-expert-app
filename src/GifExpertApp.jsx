import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {


    const [categories, setCategories] = useState([])

    return (
        <>

            {/* titulo  */}
            <div className="main-title">
                <img height={50} src="../public/gif-file.png" alt="GifExpertAllLogo"/>
                <h1>GifExpertApp</h1>

            </div>

            {/* Input Text  */}
            <AddCategory setCategories={setCategories} />

            {/* Listado de Gifs */}
            {categories.map(
                (category) => (
                    <GifGrid key={category} category={category} />
                )
            )}
            {/* gif item  */}
        </>
    )
}
