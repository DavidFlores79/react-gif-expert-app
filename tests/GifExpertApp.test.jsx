import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en GifExpertApp.jsx', () => { 
    
    test('Debe ...', () => { 
        render(<GifExpertApp/>);
        const inputValue = 'GOT';
        const input = screen.getByRole('textbox');
        fireEvent.input( input, { target: { value: inputValue } });
        screen.debug();
        expect( input.value ).toBe(inputValue);
     })

    test('Debe ser igual al snapshot', () => { 
        const { container } = render(<GifExpertApp/>);
        expect( container ).toMatchSnapshot();
     })
 })