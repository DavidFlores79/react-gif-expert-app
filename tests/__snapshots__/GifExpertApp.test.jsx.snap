// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp.jsx Debe ser igual al snapshot 1`] = `
<div>
  <div
    class="main-title"
  >
    <img
      alt="GifExpertAllLogo"
      src="../public/gif-file.png"
    />
    <h1>
      GifExpertApp
    </h1>
  </div>
  <div>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar Gifs"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
